### PR TITLE
PLG-113: Edit category properties and permissions

### PR DESCRIPTION
**Pull request by LaurentPetard**

### CODE REVIEW

File: src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/pages/categories/CategoryEditPage.tsx [hunk 1]
```diff
@@ -50,6 +53,8 @@ const CategoryEditPage: FC = () => {
   const countProductsBeforeDeleteCategory = useCountProductsBeforeDeleteCategory(parseInt(categoryId));
   const [categoryToDelete, setCategoryToDelete] = useState<CategoryToDelete | null>(null);
   const {isCategoryDeletionPossible, handleDeleteCategory} = useDeleteCategory();
+  const {setCanLeavePage} = useContext(EditCategoryContext);
+  const {editedCategory, editProperties, thereAreUnsavedChanges} = useEditCategory(category);
```
Comment (LaurentPetard) [inline src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/pages/categories/CategoryEditPage.tsx:57]:
```suggestion
  const {editedCategory, setEditedProperties, thereAreUnsavedChanges} = useEditCategory(category);
```

File: src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/pages/categories/CategoryEditPage.tsx [hunk 2]
```diff
@@ -87,6 +92,10 @@ const CategoryEditPage: FC = () => {
     setTree(rootCategory);
   }, [category]);
 
+  useEffect(() => {
```
Comment (LaurentPetard) [inline src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/pages/categories/CategoryEditPage.tsx:95]: Move to hook `useEditCategory`

File: src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/pages/categories/CategoryEditPage.tsx [hunk 3]
```diff
@@ -184,6 +215,13 @@ const CategoryEditPage: FC = () => {
             }}
           />
         )}
+        {isCurrent(permissionTabName) &&
```
Comment (olivierpontier) [inline src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/pages/categories/CategoryEditPage.tsx:218]: The permission tab should be displayed before the history tab

File: src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/hooks/categories/useCategory.ts [hunk 4]
```diff
@@ -1,19 +1,51 @@
...
+
+type FormChoiceField = FormField & {
+  value: string[];
+  choices: {
+    value: string;
+    label: string;
+  }[];
+};
+
+// @todo move to models?
```
Comment (olivierpontier) [inline src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/hooks/categories/useCategory.ts:18]: I agree, we should move those types in the Category model of our application

File: src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/pages/categories/CategoryEditPage.tsx (hunk 3 above)
Comment (LaurentPetard) [reply src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/pages/categories/CategoryEditPage.tsx:218]: It's the `<Tab.bar` that determine the order of the tabs, not this part of the code (it display  `EditPropertiesForm` or `HistoryPimView`, or  `EditPermissionsForm` but not all three at the same time)

File: src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/components/categories/EditPermissionsForm.tsx [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+          )}
+        </MultiSelectInput>
+        <Helper level="info">{translate('category.permissions.edit.help')}</Helper>
+      </Field>
+      <Field label={translate('category.permissions.own.label')}>
+        <MultiSelectInput
+          value={formData.permissions.own.value}
+          name={formData.permissions.own.fullName}
+          removeLabel="Remove"
+          emptyResultLabel="Empty result"
```
Comment (olivierpontier) [inline src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/components/categories/EditPermissionsForm.tsx:68]: Use translation

File: src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/components/categories/EditPermissionsForm.tsx [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+            </MultiSelectInput.Option>
+          )}
+        </MultiSelectInput>
+        <Helper level="info">{translate('category.permissions.edit.help')}</Helper>
+      </Field>
+      <Field label={translate('category.permissions.own.label')}>
+        <MultiSelectInput
+          value={formData.permissions.own.value}
+          name={formData.permissions.own.fullName}
+          removeLabel="Remove"
```
Comment (olivierpontier) [inline src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/components/categories/EditPermissionsForm.tsx:67]: Use translation

**Review by olivierpontier — APPROVED** [review]

File: src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/components/categories/EditPermissionsForm.tsx [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+
+  return (
+    <FormContainer>
+      <Field label={translate('category.permissions.view.label')}>
+        <MultiSelectInput
+          readOnly={false}
+          value={formData.permissions.view.value}
+          name={formData.permissions.view.fullName}
+          removeLabel="Remove"
+          emptyResultLabel="Empty result"
```
Comment (rouliane) [inline src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/components/categories/EditPermissionsForm.tsx:36]: Translations missing

**Review by rouliane — APPROVED** [review]
Units tests on complex hooks would be nice

File: src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/hooks/categories/useEditCategoryForm.ts [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+          permissions: {...editedFormData.permissions, view: {...editedFormData.permissions.view, value: values}}
+        });
+        break;
+      case 'edit':
+        setEditedFormData({
+          ...editedFormData,
+          permissions: {...editedFormData.permissions, edit: {...editedFormData.permissions.edit, value: values}}
+        });
+        break;
+      case 'edit':
```
Comment (rouliane) [inline src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/hooks/categories/useEditCategoryForm.ts:72]: Wrong case ! Should be "own"

**Pull request merged**